sound/soc/sof/sof-audio.h

```diff
@@ -42,6 +42,7 @@
 #define WIDGET_IS_DAI(id) ((id) == snd_soc_dapm_dai_in || (id) == snd_soc_dapm_dai_out)
 #define WIDGET_IS_AIF(id) ((id) == snd_soc_dapm_aif_in || (id) == snd_soc_dapm_aif_out)
 #define WIDGET_IS_AIF_OR_DAI(id) (WIDGET_IS_DAI(id) || WIDGET_IS_AIF(id))
+#define WIDGET_IS_COPIER(id) (WIDGET_IS_AIF_OR_DAI(id) || (id) == snd_soc_dapm_buffer)
```

Comment:
paranoia check: is there a 1:1 mapping between dapm_buffer type and copier? Can this type be selected by anything else?

Reply:
the dapm_buffer type can only be used for copier, no other components
````c
	[snd_soc_dapm_buffer] = {sof_ipc4_widget_setup_pcm, sof_ipc4_widget_free_comp_pcm,
				 common_copier_token_list, ARRAY_SIZE(common_copier_token_list),
				 NULL, sof_ipc4_prepare_copier_module,
				 sof_ipc4_unprepare_copier_module},
````

Reply:
> paranoia check: is there a 1:1 mapping between dapm_buffer type and copier? Can this type be selected by anything else?

For IPC4, yes. But IPC3 has buffer type widget. Maybe rename it to  WIDGET_IS_IPC4_COPIER(id)?